client/src/pages/SignUp.js

```diff
@@ -0,0 +1,238 @@
+import React, { useState } from "react";
+import {
+  Button,
+  Typography,
+  TextField,
+  Card,
+  CardContent,
+  Grid,
+} from "@material-ui/core";
+import { Link } from "react-router-dom";
+import { withStyles } from "@material-ui/core/styles";
+import BackgroundImg from "../Assets/images/signUpBkg.png";
+import ChefsMenuLogo from "../Assets/images/Logo.png";
+
+const regPattern = new RegExp(/^[\w-.]+@([\w-]+\.)+[\w-]{2,4}$/i);
+
+const signUpPageStyle = (theme) => ({
+  signUpContainer: {
+    fontFamily: '"Montserrat"',
+    margin: theme.spacing(2),
+    height: "100vh",
+  },
+  title: {
+    fontSize: 25,
+    fontWeight: "bold",
+    textAlign: "center",
+  },
+  button: {
+    color: "white",
+    width: theme.spacing(15),
+    background: "#FF510C", //Brighter Orange
+    "&:hover": {
+      backgroundColor: "#FF510C", //Brighter Orange
+    },
+  },
+  titleLogo: {
+    width: "50%",
+    height: "10vh",
+    backgroundImage: `url(${ChefsMenuLogo})`,
+    backgroundSize: "contain",
+    backgroundRepeat: "no-repeat",
+  },
+  rightSideContainer: {
+    [theme.breakpoints.up("md")]: {
+      backgroundImage: `url(${BackgroundImg})`,
+      backgroundSize: "cover",
+      backgroundPosition: "left",
+    },
+  },
+  textField: {
+    margin: theme.spacing(2),
+    width: "30ch",
+  },
+  textColor: {
+    [theme.breakpoints.up("md")]: {
+      color: "white",
+    },
+    color: "black",
+  },
+  marginLeft_2: {
+    margin: "2em",
+  },
+  linkNoDecoration: {
+    textDecoration: "none",
+  },
+});
+
+const SignUpPage = (props) => {
+  const [username, setUserName] = useState("");
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [usernameError, setUserNameError] = useState("");
+  const [emailError, setEmailError] = useState("");
+  const [passwordError, setPasswordError] = useState("");
+  const [passwordConfirmError, setPasswordConfirmError] = useState("");
+
+  const { classes } = props;
+
+  const validateChange = (e) => {
+    switch (e.target.name) {
```

Comment:
I think this is fine for now, but in the future we should refactor this to not use a switch statement. Something more like:
```javascript
const userUpdate = {
  ...user,
  [e.target.name]: e.target.value
}
```